src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -3749,6 +3749,15 @@ theorem two_mul {x : BitVec w} : 2#w * x = x + x := by rw [BitVec.mul_comm, mul_
   (x * y).toInt = (x.toInt * y.toInt).bmod (2^w) := by
   simp [toInt_eq_toNat_bmod, -Int.natCast_pow]
 
+@[simp]
+theorem toInt_mul_of_not_smulOverflow {x y : BitVec w} (h : ¬ smulOverflow x y) :
+    (x * y).toInt = x.toInt * y.toInt := by
+  rcases w with _|w
+  · simp [of_length_zero]
+  · simp only [smulOverflow, Nat.add_one_sub_one, ge_iff_le, Bool.or_eq_true, decide_eq_true_eq,
+    _root_.not_or, Int.not_le, Int.not_lt] at h
+    simp [Int.bmod_eq_of_le (n := x.toInt * y.toInt) (m := 2 ^ (w + 1)) (by push_cast; omega) (by push_cast; omega)]
```

Comment:
```suggestion
    simp [Int.bmod_eq_of_le (n := x.toInt * y.toInt) (m := 2 ^ (w + 1)) (by push_cast; omega)
      (by push_cast; omega)]
```

The line was over the 120 character limit.

Reply:
Also, if you replace the simp with `rw` you, may not need to specify the arguments explicitly.